### PR TITLE
drarm: Add DrArm workload

### PR DESCRIPTION
**Pull request by mrkajetanp**
Add a workload for the Dr Arm demo app. Includes functionality for automatically pulling the ADPF FPS report file from the target if one was generated by the app.

### CODE REVIEW

File: wa/workloads/drarm/__init__.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        if self.fixed_time_step is not None:
+            args['fixedTimeStep'] = self.fixed_time_step
+
+        if self.fps_report_file is not None:
+            args['fpsReportFileName'] = self.fps_report_file
+
+        return args
+
+    def run(self, context):
+        self.target.sleep(self.timeout)
```
Comment (marcbonnici) [inline wa/workloads/drarm/__init__.py:82]: Does this workload not need to do anything in the run stage? Usually we want the workload to be kicked off here so the operation that is measured is consistent across runs.

File: wa/workloads/drarm/__init__.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    def update_output(self, context):
+        super(DrArm, self).update_output(context)
+        outfile_glob = self.target.path.join(self.target.persistent_data_path, self.apk.package, 'files', '*.csv')
+
+        try:
+            ls_output = self.target.execute('ls {}'.format(outfile_glob), as_root=True)
+        except Exception:
```
Comment (marcbonnici) [inline wa/workloads/drarm/__init__.py:90]: Can you be more specific in the exception here. Should this be a `TargetStableError` etc.?

File: wa/workloads/drarm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        on_target_output_files = [f.strip() for f in ls_output.split('\n') if f]
+
+        self.logger.info('Extracting the ADPF FPS report from target...')
+        for file in on_target_output_files:
+            host_output_file = os.path.join(context.output_directory, os.path.basename(file))
+            self.target.pull(file, host_output_file)
+            context.add_artifact('adpf', host_output_file, kind='data',
+                                 description='ADPF report log in CSV format.')
+
+
```
Comment (marcbonnici) [inline wa/workloads/drarm/__init__.py:103]: nit from linting: Extra new line.

File: wa/workloads/drarm/__init__.py (hunk 2 above)
Comment (mrkajetanp) [reply wa/workloads/drarm/__init__.py:90]: Good point, thanks, it should be `TargetStableCalledProcessError`

File: wa/workloads/drarm/__init__.py (hunk 1 above)
Comment (mrkajetanp) [reply wa/workloads/drarm/__init__.py:83]: The build we have has a command line option (`autoDemo`) that just starts the workload automatically a few seconds after the app is opened so we don't need to do anything in the run stage explicitly. It just always starts in the same way from the get-go.

**Comment by mrkajetanp** [comment]
Updated in line with changes to https://github.com/ARM-software/devlib/pull/622

File: wa/workloads/drarm/__init__.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+class DrArm(ApkWorkload):
+
+    name = 'drarm'
+    package_names = ['com.Arm.DrArm']
+    activity = 'com.unity3d.player.UnityPlayerActivity'
+    install_timeout = 200
+    description = """
+    Dr. Arm’s Amazing Adventures is a “Souls-Like” Mobile Action Role Playing Game developed at Arm.
```
Comment (marcbonnici) [inline wa/workloads/drarm/__init__.py:30]: Is this app publicly available? If so would it be possible to provide a link to where the application can be obtained from?

File: wa/workloads/drarm/__init__.py (hunk 1 above)
Comment (marcbonnici) [reply wa/workloads/drarm/__init__.py:85]: Is it worth executing that command to kick the workload off in the run stage itself? That way any results wouldn't be affected by the number/which instruments were also enabled on a given run?

File: wa/workloads/drarm/__init__.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            args['fpsReportFileName'] = self.fps_report_file
+
+        return args
+
+    def run(self, context):
+        self.target.sleep(self.timeout)
+
+    def update_output(self, context):
+        super(DrArm, self).update_output(context)
+        outfile_glob = self.target.path.join(self.target.external_storage_app_dir, self.apk.package, 'files', '*.csv')
```
[the remark this answers is not in the document]
Comment (mrkajetanp) [reply wa/workloads/drarm/__init__.py]: sure, done

File: wa/workloads/drarm/__init__.py (hunk 4 above)
Comment (mrkajetanp) [reply wa/workloads/drarm/__init__.py:30]: Hm I don't think so, it's an Arm-internal demo. It's been shown publicly like here:

https://community.arm.com/arm-community-blogs/b/graphics-gaming-and-vr-blog/posts/how-to-use-adaptive-performance

But we don't have the actual app published anywhere outside the internal systems as far as I know

**Pull request merged**